lib/ajax-datatables-rails/datatable/column/search.rb

```diff
@@ -82,6 +89,18 @@ def numeric_search
           end
         end
 
+        def is_searchable_integer?
+          return true unless table.respond_to?(:engine)
+          table.engine.columns_hash[field.to_s].sql_type == 'integer' && is_integer?(search.value) && !is_out_of_range?(search.value)
```

Comment:
You get to this point on Rails 4 but `is_out_of_range?` always return false because it (can) receives an array of integer and `is_integer?(string)` shadows the casting error by rescuing to false.
```
  1) AjaxDatatablesRails::ORM::ActiveRecord filter conditions it can filter records with condition :in should filter records matching
     Failure/Error: Integer(search_value) > LARGEST_PQ_INTEGER || Integer(search_value) < SMALLEST_PQ_INTEGER
     
     TypeError:
       can't convert Array into Integer
```